internal/provider/provider_test.go

```diff
@@ -44,26 +53,173 @@ var testAccProtoV6ProviderFactories = map[string]func() (tfprotov6.ProviderServe
 	// globalAnnotation is explicitly set here, as the provider initialization in acceptance tests differs from the standard initialization process.
 	setGlobalAnnotation(basetypes.NewStringNull(), "ACI_ANNOTATION")
 
-	if v := os.Getenv("ACI_USERNAME"); v == "" {
-		t.Fatal("ACI_USERNAME must be set for acceptance tests")
+	if testType == "cloud" {
+		t.Skip("[WARNING] Test skipped because it is not supported on an on-prem APIC")
+	}
+
+	infoController, err := getACIClientTest(t).GetViaURL("/api/node/class/firmwareCtrlrRunning.json")
+	if err != nil {
+		t.Fatalf("Error fetching APIC controller information: %v", err)
+	}
+	apicVersion := extractControllerVersion(infoController)
+	// TODO process the version when it has ranges associated with it
+	if apicVersion != strings.TrimSuffix(testApplicableFromVersion, "-") {
+		if auxillaryFunctions.IsVersionGreater(*auxillaryFunctions.ParseVersion(testApplicableFromVersion).Version, *auxillaryFunctions.ParseVersion(apicVersion).Version) {
+			t.Skip("[WARNING] Test skipped because it is not supported on APIC version:", apicVersion)
+		}
+	}
+
+}
+
+func ComposeAggregateTestCheckFuncWithVersion(t *testing.T, propertyVersion string, operator string, checks ...resource.TestCheckFunc) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources["data.aci_system.version"]
+
+		if !ok {
+			return fmt.Errorf("data source aci_system was not found in the test configuration")
+		}
+
+		apicVersion := rs.Primary.Attributes["version"]
+		apicVersionResult := auxillaryFunctions.ParseVersion(apicVersion)
+
+		if apicVersionResult.Error == "unknown" {
+			return fmt.Errorf("Error reading version from the apic in the test check function: %s", apicVersion)
+		}
+
+		apicV := *apicVersionResult.Version
+
+		propertyVersionResult := auxillaryFunctions.ParseVersion(propertyVersion)
+		if propertyVersionResult.Error == "unknown" {
+			return nil
+		}
+		propertyV := *propertyVersionResult.Version
+
+		var result []error
+		var comparisonResult bool
+
+		switch operator {
+		case "==":
+			comparisonResult = auxillaryFunctions.IsVersionEqual(propertyV, apicV)
+		case "!=":
+			comparisonResult = auxillaryFunctions.IsVersionNotEqual(propertyV, apicV)
+		case ">":
+			comparisonResult = auxillaryFunctions.IsVersionGreater(propertyV, apicV)
+		case "<":
+			comparisonResult = auxillaryFunctions.IsVersionLesser(propertyV, apicV)
+		case ">=":
+			comparisonResult = auxillaryFunctions.IsVersionGreaterOrEqual(propertyV, apicV)
+		case "<=":
+			comparisonResult = auxillaryFunctions.IsVersionLessOrEqual(propertyV, apicV)
+		default:
+			return fmt.Errorf("Invalid operator found in the test check function: %s", operator)
+		}
+
+		if comparisonResult {
+			t.Logf("[WARNING] Test for checking the resource attribute value skipped due to version incompatibility between Property version: %s and APIC version: %s", propertyVersion, apicVersion)
```

Comment:
Can we add the variable name of the resource in the error?

Reply:
As discussed, this will not be implemented now.